File: src/main/scala/io/lamma/Month.scala
package io.lamma

sealed trait Month {
  def ordinal = Month.Months.indexOf(this) + 1
}

object Month {
  /**
   * get month from int, starting from 1
   * @param n
   * @return
   */
  def apply(n: Int): Month = Months(n - 1)

  final val Months = List(January, February, March, April, May, June, July, August, September, October, November, December)
}

case object January extends Month

case object February extends Month

case object March extends Month

case object April extends Month

case object May extends Month

case object June extends Month

case object July extends Month

case object August extends Month

case object September extends Month

case object October extends Month

case object November extends Month

case object December extends Month

File: src/main/java/io/lamma/Month.java
package io.lamma;

public enum Month {

    JANUARY(Quarter.Q1),

    FEBRUARY(Quarter.Q1),

    MARCH(Quarter.Q1),

    APRIL(Quarter.Q2),

    MAY(Quarter.Q2),

    JUNE(Quarter.Q2),

    JULY(Quarter.Q3),

    AUGUST(Quarter.Q3),

    SEPTEMBER(Quarter.Q3),

    OCTOBER(Quarter.Q4),

    NOVEMBER(Quarter.Q4),

    DECEMBER(Quarter.Q4);

    private final Quarter q;

    Month(Quarter q) {
        this.q = q;
    }

    public Quarter quarter() {
        return this.q;
    }

    public int n() {
        return this.ordinal() + 1;
    }

    public static Month of(int i) {
        return values()[i - 1];
    }

}
